### PR TITLE
Various updates for NOT ALFOSC

### PR DESCRIPTION
**Pull request by Raindogjones**
PR adds:

- reid_arxiv for two more grisms (8 and 17)
- A class to handle vertical slits (not_alfosc_vert)
- A minimum slit separation to avoid the misidentification of multiple slits due to dust on the slit masks (ALFOSC is not a multislit instrument anyway)
- Rudimentary documentation
- Changes the way arcs (from different lamps often with different exposure times) are combined to be in line with the implementation for GTC OSIRIS

### CODE REVIEW

File: pypeit/spectrographs/not_alfosc.py [hunk 1]
```diff
@@ -111,21 +111,32 @@ def default_pypeit_par(cls):
...
+        # Not a multi-slit instrument but flats are sometimes quite ugly leading to erroneous detection of multiple slits. Setting a higher edge_thresh seems to fail because of the bound_detector.
+        par['calibrations']['slitedges']['minimum_slit_gap'] = 15
 
         # Set pixel flat combination method
         par['calibrations']['pixelflatframe']['process']['combine'] = 'median'
         # Wavelength calibration methods
         #par['calibrations']['wavelengths']['method'] = 'holy-grail'
         par['calibrations']['wavelengths']['method'] = 'full_template'
-        par['calibrations']['wavelengths']['lamps'] = ['HeI', 'NeI']
+        par['calibrations']['wavelengths']['lamps'] = ['HeI', 'NeI', 'ThAr']
```
Comment (rcooke-ast) [inline pypeit/spectrographs/not_alfosc.py]: This makes me a little nervous... The ThAr list contains 17000 entries, while the NOT/ALFOSC ThAr list is only about 100 entries. My feeling is that you might be able to get away with adding `'ArI', 'ArII'` to this list, and possibly supplement this with a list of `'ThI'` lines based on the list provided on [NOT/ALFOSC page](http://www.not.iac.es/instruments/alfosc/lamps/thar.txt)?

File: pypeit/spectrographs/not_alfosc.py [hunk 2]
```diff
@@ -28,7 +28,7 @@ class NOTALFOSCSpectrograph(spectrograph.Spectrograph):
     url = 'https://www.not.iac.es/instruments/alfosc/'
     header_name = 'ALFOSC_FASU'
     supported = True
-    comment = 'Grisms 4, 19'
+    comment = 'Grisms 4, 8, 17, 19'
```
Comment (rcooke-ast) [inline pypeit/spectrographs/not_alfosc.py]: Maybe with the comment here you could add "For use with the horizontal slit mode only."

File: pypeit/spectrographs/not_alfosc.py [hunk 3]
```diff
@@ -269,3 +284,85 @@ def config_specific_par(self, scifile, inp_par=None):
...
         return par
 
+
+
+
+class NOTALFOSCSpectrographVert(NOTALFOSCSpectrograph):
+    """
+    Child to handle Vertical slits for NOT ALFOSC spectrograph
+    """
+    ndet = 1
```
Comment (rcooke-ast) [inline pypeit/spectrographs/not_alfosc.py]: Most of these will be inherited (I think?!). I think the only information that needs to be updated here is:
```
    name = 'not_alfosc_vert'
    comment = 'Grisms 4, 8, 17, 19. For vertical slits only.'
```

File: pypeit/spectrographs/not_alfosc.py [hunk 4]
```diff
@@ -269,3 +284,85 @@ def config_specific_par(self, scifile, inp_par=None):
...
+            saturation      = 700000.,  # ADU
+            nonlinear       = 0.86,
+            datasec         = np.atleast_1d('[{}:{},:]'.format(1, 2148)),  # Unbinned
+            oscansec        = None,
+            numamplifiers   = 1,
+            gain            = gain,     # e-/ADU
+            ronoise         = ronoise   # e-
+        )
+
+#        # Parse datasec, oscancsec from the header
```
Comment (rcooke-ast) [inline pypeit/spectrographs/not_alfosc.py]: I think these comment lines can be removed

**Review by rcooke-ast — COMMENTED** [review]
All looks good to me. Mostly just a few minor things. Do you think it would be beneficial to add something to the development suite?

File: pypeit/spectrographs/not_alfosc.py (hunk 1 above)
Comment (Raindogjones) [reply pypeit/spectrographs/not_alfosc.py]: I've just tested it for grism 17 (which is the only one of the four that really uses the ThAr lamp) and you are right, we get a good, and less risky, wavelength solution just using ArI and ArII.  I will omit ThI for the moment but think about including a line list when I implement the rest of the grisms.

**Comment by Raindogjones** [comment]
The only data I currently have access to are still in their proprietary period and belong to someone else but, once I get my hands on the data for the other grisms, I'll make sure to add a handful of datasets to the development suite (in particular making sure to get at least a couple with vertical slits).

File: doc/spectrographs/not_alfosc.rst [hunk 5]
```diff
@@ -0,0 +1,23 @@
+==========
```
Comment (profxj) [inline doc/spectrographs/not_alfosc.rst:1]: Please add this to the list in `doc/spectrographs/spectrographs.rst`

File: pypeit/spectrographs/not_alfosc.py [hunk 6]
```diff
@@ -111,21 +111,32 @@ def default_pypeit_par(cls):
...
+        # Not a multi-slit instrument but flats are sometimes quite ugly leading to erroneous detection of multiple slits. Setting a higher edge_thresh seems to fail because of the bound_detector.
+        par['calibrations']['slitedges']['minimum_slit_gap'] = 15
 
         # Set pixel flat combination method
         par['calibrations']['pixelflatframe']['process']['combine'] = 'median'
         # Wavelength calibration methods
         #par['calibrations']['wavelengths']['method'] = 'holy-grail'
         par['calibrations']['wavelengths']['method'] = 'full_template'
-        par['calibrations']['wavelengths']['lamps'] = ['HeI', 'NeI']
+        par['calibrations']['wavelengths']['lamps'] = ['HeI', 'NeI', 'ArI', 'ArII']
```
Comment (profxj) [inline pypeit/spectrographs/not_alfosc.py:122]: Are these lamps good for all the gratings or only the new ones?

**Review by profxj — CHANGES_REQUESTED** [review]
Thanks for these additions!

I have a few minor requests/questions.

File: pypeit/spectrographs/not_alfosc.py (hunk 6 above)
Comment (Raindogjones) [reply pypeit/spectrographs/not_alfosc.py:122]: I believe the calibration plan is to provide single He, Ne and ThAr exposures for each setup used - http://www.not.iac.es/observing/seq/new/seq-command.php?i=alfosc.alfosc-calibs - meaning they should be good for all grisms.  I intend to implement the rest of the grisms once I have access to the relevant data, so I'll submit a future pull request to make this a setup dependent parameter if I discover that isn't the case.

**Comment by profxj** [comment]
I can confirm that tests pass on the DevSuite *provided* 
one pulls in `develop`.  Not sure why they fail without (some
edge of flat problems)..

**Comment by Raindogjones** [comment]
After pulling in `develop` the reduction runs without throwing an error but doesn't extract any spectra.  Seems to be something going wrong with the sky subtraction as the skysub frame is all zeros.

**Comment by Raindogjones** [comment]
> After pulling in `develop` the reduction runs without throwing an error but doesn't extract any spectra. Seems to be something going wrong with the sky subtraction as the skysub frame is all zeros.

Mea culpa.  Was me not performing a fresh install of pypeit.  Everything seems to be working as it should now.

**Comment by profxj** [comment]
I have verified the DevSuite runs and the wavelength solutions
for grism4 and grism19 are fine.

Am approving.  Unit test failures are unrelated to this PR

**Review by profxj — APPROVED** [review]

**Review by rcooke-ast — APPROVED** [review]
Approved!

**Pull request merged**